saltstack/base/salt/prerequisites/usr/bin/ccmv2-helper.sh

```diff
@@ -33,12 +34,16 @@ setup_ccmv2() {
 
   INVERTING_PROXY_URL="$CCM_V2_INVERTING_PROXY_HOST"
 
+  CCM_V2_AGENT_ACCESS_KEY_ID=$(grep '^CCM_V2_AGENT_ACCESS_KEY_ID=' ${ACC_KEY_TMP_FILE} | cut -d'=' -f2)
+
   if [[ "$IS_CCM_V2_JUMPGATE_ENABLED" == "true" && "$IS_FREEIPA" == "true" && ! -z $CCM_V2_AGENT_ACCESS_KEY_ID ]]; then
-    ACCESS_KEY_ID="$CCM_V2_AGENT_ACCESS_KEY_ID"
+
+    CCM_V2_AGENT_ENCIPHERED_ACCESS_KEY=$(grep '^CCM_V2_AGENT_ENCIPHERED_ACCESS_KEY=' ${ACC_KEY_TMP_FILE} | cut -d'=' -f2)
 
     if [[ -z $CCM_V2_AGENT_HMAC_KEY ]]; then
       # legacy without HMAC check, AES-128
       ACCESS_KEY="$(echo ${CCM_V2_AGENT_ENCIPHERED_ACCESS_KEY} | openssl enc -aes-128-cbc -d -A -a -K ${CCM_V2_AGENT_KEY_HEX} -iv ${LEGACY_IV})"
+      echo "ACCESS_KEY=$ACCESS_KEY" >> $ACC_KEY_TMP_FILE
```

Comment:
I fear these (at least the `echo`) might reveal the access key in the log, which would be nice to avoid.

Reply:
actually, it shouldn't, because in the caller script the `set -x` got set up thus no detailed logging happens neither there nor here

Reply:
Have you actually tried this? :smile: `set -x` enables command logging; `set +x` disables it.